lib/index.ts

```diff
@@ -16,23 +16,33 @@
  */
 
 import * as _ from 'lodash';
+import type * as Stream from 'stream';
 
-import type { DockerVersion, ImagePushOptions } from 'dockerode';
+import type {
+	DockerVersion,
+	ImagePushOptions,
+	ImageBuildOptions,
+	AuthConfig,
+} from 'dockerode';
 
 export interface EngineVersion extends DockerVersion {
 	Engine?: string;
 }
 
 export interface PullPushOptions {
-	authconfig?: {
-		username?: string;
-		password?: string;
+	ignoreProgressErrorEvents?: boolean;
+	authconfig?: AuthConfig & {
 		auth?: string;
-		email?: string;
-		serveraddress?: string;
 		registrytoken?: string;
 	};
+}
+
+export interface BuildOptions extends ImageBuildOptions {
 	ignoreProgressErrorEvents?: boolean;
+	authconfig?: AuthConfig & {
+		auth?: string;
+		registrytoken?: string;
+	};
```

Comment:
Defining the type like this allows it to extend from `ImageBuildOptions` and be used as an argument to `docker.buildImage` and also allow for extra configurations.